app/javascript/App/bundles/ExchangeListings/client.js

```diff
@@ -153,27 +154,30 @@ class ExchangeListingsPage extends Component {
       detectedUntil: this.state.detectedUntil,
     }
 
+    const context = {
+      toggleFilterPanel: this.toggleFilterPanel,
+      showFilterPanel: this.state.showFilterPanel,
+      applyFilters: () => this.applyFilters(),
+      quoteSymbols: this.props.quoteSymbols,
+      exchanges: this.props.exchanges,
+      changeSymbol: this.changeSymbol,
+      changeExchange: this.changeExchange,
+      filterDates: this.filterDates,
+      selectedItems: selectedItems,
+      selectedSymbols: this.state.selectedSymbols,
+      selectedExchanges: this.state.selectedExchanges,
+      exchangeSlugs: this.state.exchangeSlugs,
+      resetFilters: this.resetFilters,
+    }
+
     if (window.isMobile) {
       return (
         <LayoutMobile
           mainSection={
             <Fragment>
-              <ListingsHeader
-                toggleFilterPanel={this.toggleFilterPanel}
-                showFilterPanel={this.state.showFilterPanel}
-                quoteSymbols={this.props.quoteSymbols}
-                exchanges={this.props.exchanges}
-                toggleFilterPanel={this.toggleFilterPanel}
-                applyFilters={() => this.applyFilters()}
-                changeSymbol={this.changeSymbol}
-                changeExchange={this.changeExchange}
-                filterDates={this.filterDates}
-                selectedItems={selectedItems}
-                selectedSymbols={this.state.selectedSymbols}
-                selectedExchanges={this.state.selectedExchanges}
-                exchangeSlugs={this.state.exchangeSlugs}
-                resetFilters={this.resetFilters}
-              />
+              <Provider value={context}>
+                <ListingsHeader showFilterPanel={this.state.showFilterPanel} />
```

Comment:
this property is the only one that is used at that level. the other 12 were just being passed through.

Reply:
Nice, great cleanup! ✨